tests/AnalyticsEventReporter.test.ts

```diff
@@ -359,27 +310,11 @@ describe('Test report function', () => {
 
   it('calling report with no argument and no with call should result in invalid request body and error',
     async () => {
-      mockPostWithFetch.mockResolvedValue({
-        ok: false,
-        status: 400,
-        statusText: 'Bad Request',
-        json: jest.fn().mockResolvedValue({}),
-        redirected: false,
-        type: 'basic',
-        url: 'https://example.com',
-        clone: jest.fn(),
-        headers: new Headers(),
-        body: null,
-        bodyUsed: false,
-        arrayBuffer: jest.fn(),
-        blob: jest.fn(),
-        formData: jest.fn(),
-        text: jest.fn()
-      });
+      mockPostWithBeacon.mockReturnValue(false);
+      mockPostWithFetch.mockResolvedValue({errors: ['Bad Request']});
```

Comment:
Do we ever mock the failed promise? i only see us mockPostWithFetch with mockResolvedValue here in this file.

Reply:
Ah these should be using mockRejectedValue